unified-codes/apps/web/src/components/containers/detail/DetailEntityListItem.tsx

```diff
@@ -94,10 +96,21 @@ const DetailEntityListItemComponent: DetailEntityListItem = ({
     if (!childCount) return null;
     const childItems = childEntities?.map((child: IEntity) => {
       const { description, children, type } = child;
-      const childForm = type === 'form' ? description : undefined;
+      const childForm = type === 'form' ? description.toLowerCase() : undefined;
+      let fullDescription = description;
+      let parentDescription = entityDescription;
+      switch(type) {
+        case 'form':
+          parentDescription = `${entityDescription} ${description}`;
+          break;
+        case 'unit_of_use':
+          fullDescription =  `${entityDescription} ${description}`;
+          break;
+      }
```

Comment:
If I understand this right, this propagates higher level descriptions down to lower levels (and then concatenates them). I can understand the motivation, but the solution here breaks the purity of the recursion and makes it a bit confusing (at least to me!). Would an alternative (and possibly simpler) idea be to update the data (either at the `.csv`/spreadsheet stage or just moving this logic out of the UI and into the pre-processing), rather than prop drilling the parent's description? The benefit of manually setting the descriptions in the spreadsheet is that it avoid annoying edge-cases and such which could bloat this logic more and more (I'm guessing there will be situations where concatenating the descriptions might not give the desired effect)?

Reply:
😄  yes, it's awful.
Are you suggesting we return to having a complete description on each entity rather than splitting out all the elements?

Reply:
Selectors!
I thought maybe reformatting as selectors would be nice; `getEntityDescription( code )` could return a fully formatted description

Reply:
Yeah, that way the `description` can be passed directly into the component. I think `selectEntityDescription` would be the preferred way of doing it, for consistency with other selectors (see `src/selectors/DetailSelectors`).